packages/apps/reputation-oracle/server/src/modules/cron-job/cron-job.service.ts

```diff
@@ -85,91 +96,192 @@ export class CronJobService {
 
   @Cron('*/2 * * * *')
   /**
-   * Process a pending webhook job.
-   * @returns {Promise<void>} - Returns a promise that resolves when the operation is complete.
+   * Processes all pending incoming webhooks, marking them as completed upon success.
+   * Handles any errors by logging them and updating the webhook status.
+   * @returns {Promise<void>} A promise that resolves when all pending incoming webhooks have been processed.
    */
-  public async processPendingWebhooks(): Promise<void> {
+  public async processPendingIncomingWebhooks(): Promise<void> {
     const isCronJobRunning = await this.isCronJobRunning(
-      CronJobType.ProcessPendingWebhook,
+      CronJobType.ProcessPendingIncomingWebhook,
     );
 
     if (isCronJobRunning) {
       return;
     }
 
-    this.logger.log('Pending webhooks START');
-    const cronJob = await this.startCronJob(CronJobType.ProcessPendingWebhook);
+    this.logger.log('Pending incoming webhooks START');
+    const cronJob = await this.startCronJob(
+      CronJobType.ProcessPendingIncomingWebhook,
+    );
 
     try {
-      const webhookEntities = await this.webhookRepository.findByStatus(
-        WebhookStatus.PENDING,
+      const webhookEntities = await this.webhookIncomingRepository.findByStatus(
+        WebhookIncomingStatus.PENDING,
       );
 
       for (const webhookEntity of webhookEntities) {
-        let resultsUrl;
         try {
           const { chainId, escrowAddress } = webhookEntity;
-          resultsUrl = await this.payoutService.executePayouts(
+
+          await this.webhookService.createEscrowCompletionTracking(
             chainId,
             escrowAddress,
           );
         } catch (err) {
-          this.logger.error(`Error sending webhook: ${err.message}`);
-          await this.webhookService.handleWebhookError(webhookEntity);
+          const errorId = uuidv4();
+          const failedReason = `${ErrorWebhook.PendingProcessingFailed} (Error ID: ${errorId})`;
```

Comment:
Make sense 
Correct, since we store the `errorId` in the database and include the error message in the logs, we need to copy the `errorId` and find the error message. Of course, we could store the message in the database, but would that be redundant? I think at least it would be worth adding a limit on the length of the message being stored.

I would suggest implementing this in a separate PR as a part of refactoring.
@Dzeranov @dnechay

Reply:
The confusing part is that it's called `failedReason`, but doesn't have anything related to a reason at all.

> Of course, we could store the message in the database, but would that be redundant?
If we store only error message, not stack, that wouldn't be, but might not be informative enough.

Taking the above into account - I would go simplest path and rename this field to something like `failureDetail` to avoid confusion and not change the format (should not take long taking into account we already migrate tables and keep the data type). Later we can think about better approach if there is a need